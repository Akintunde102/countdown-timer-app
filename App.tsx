import React, {useEffect, useState, useContext} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  StatusBar,
  KeyboardAvoidingView,
} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {Timer, InputForm} from './components';
import {SizeContextProvider, SizeContext} from './contexts';

const App = () => {
  const {dHeight, dWidth} = useContext(SizeContext);
  const [duration, setDuration] = useState<number>(0);
  const [timerStatus, setTimerStatus] = useState<{
    starts: boolean;
    time?: Date;
  }>({
    starts: false,
  });
  useEffect(() => {
    SplashScreen.hide();
  });

  const styles = StyleSheet.create({
    app: {
      height: dHeight,
      width: dWidth,
      borderTopColor: '#000',
      borderTopWidth: 1,
    },
    inputForm: {
      flexDirection: 'row',
      borderBottomWidth: 1,
      borderRadius: 2,
      borderColor: '#ddd',
      shadowColor: '#000',
      shadowOffset: {width: 0, height: 2},
      shadowOpacity: 0.8,
      shadowRadius: 2,
      elevation: 5,
    },
    timer: {},
  });

  return (
    <>
      <StatusBar
        barStyle="dark-content"
        hidden={false}
        backgroundColor="#fff"
      />
      <SafeAreaView>
        <KeyboardAvoidingView behavior="height">
          <View style={styles.app}>
            <SizeContextProvider>
              <InputForm
                setDuration={setDuration}
                setTimerStatus={setTimerStatus}
                style={styles.inputForm}
              />
              <Timer
                style={styles.timer}
                startDetails={timerStatus}
                durationInMin={duration}
              />
            </SizeContextProvider>
          </View>
        </KeyboardAvoidingView>
      </SafeAreaView>
    </>
  );
};

export default App;
