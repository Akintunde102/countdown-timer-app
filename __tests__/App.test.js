import 'react-native';
import React from 'react';
import App from '../App.tsx';
import renderer from 'react-test-renderer';

jest.mock('react-native-splash-screen', () => {
  return {
    hide: jest.fn(),
    show: jest.fn(),
  };
});

jest.useFakeTimers();

it('App Snapshot', () => {
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
});
