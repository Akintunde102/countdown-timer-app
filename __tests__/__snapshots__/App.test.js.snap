// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "borderTopColor": "#000",
          "borderTopWidth": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffc",
            "borderBottomWidth": 1,
            "borderColor": "#ddd",
            "borderRadius": 2,
            "elevation": 5,
            "flexDirection": "row",
            "height": 106.72,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 2,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "flex": 1,
              "justifyContent": "center",
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignContent": "center",
                "fontSize": 37.5,
                "fontWeight": "700",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            CountDown:
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "flex": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="(Min)"
            placeholderTextColor="#777776"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#62bfac",
                "fontSize": 50,
                "fontStyle": "normal",
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#62bfac",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 52.083333333333336,
                "textAlign": "center",
              }
            }
          >
            START
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ffc",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "marginBottom": 13.34,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 266.8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.1,
                "justifyContent": "center",
                "marginRight": 75,
                "textAlign": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 41.66666666666667,
                  "fontWeight": "700",
                }
              }
            >
               
              
               
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 0.4,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 6,
                  "justifyContent": "center",
                  "textAlign": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "display": "none",
                    "fontFamily": "uFont",
                    "fontSize": 241.66666666666669,
                    "textAlign": "center",
                  }
                }
              >
                00:00
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 37.5,
                  "padding": 0,
                  "textAlign": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 93.75,
                      },
                      Object {},
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  î€¹
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderColor": "#ddd",
              "borderRadius": 2,
              "borderTopWidth": 10,
              "flex": 0.2,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6c6c6c",
                "borderColor": "#000",
                "borderWidth": 3,
                "justifyContent": "center",
                "marginBottom": 40.019999999999996,
                "marginLeft": 0,
                "marginRight": 22.5,
                "marginTop": 26.68,
                "opacity": 1,
                "paddingBottom": 40.019999999999996,
                "paddingLeft": 22.5,
                "paddingRight": 22.5,
                "paddingTop": 40.019999999999996,
                "textAlign": "center",
                "width": 180,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 41.66666666666667,
                  "fontWeight": "700",
                }
              }
            >
              1X
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#000",
                "borderWidth": 3,
                "justifyContent": "center",
                "marginBottom": 40.019999999999996,
                "marginLeft": 0,
                "marginRight": 22.5,
                "marginTop": 26.68,
                "opacity": 1,
                "paddingBottom": 40.019999999999996,
                "paddingLeft": 22.5,
                "paddingRight": 22.5,
                "paddingTop": 40.019999999999996,
                "textAlign": "center",
                "width": 180,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 41.66666666666667,
                  "fontWeight": "700",
                }
              }
            >
              1.5X
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#000",
                "borderWidth": 3,
                "justifyContent": "center",
                "marginBottom": 40.019999999999996,
                "marginLeft": 0,
                "marginRight": 22.5,
                "marginTop": 26.68,
                "opacity": 1,
                "paddingBottom": 40.019999999999996,
                "paddingLeft": 22.5,
                "paddingRight": 22.5,
                "paddingTop": 40.019999999999996,
                "textAlign": "center",
                "width": 180,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 41.66666666666667,
                  "fontWeight": "700",
                }
              }
            >
              2X
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
